packages/flow-remove-types/index.js

```diff
@@ -98,32 +101,34 @@ function resultPrinter(options, source, removedNodes) {
         return source;
       }
 
-      var result = '';
+      var buf = Buffer.alloc(sourceBuffer.length + bytesAdded);
       var lastPos = 0;
+      var offset = 0;
 
       // Step through the removed nodes, building up the resulting string.
       for (var i = 0; i < removedNodes.length; i++) {
         var node = removedNodes[i];
-        result += source.slice(lastPos, startOf(node));
+        offset += sourceBuffer.copy(buf, offset, lastPos, startOf(node));
         lastPos = endOf(node);
         if (typeof node.__spliceValue === 'string') {
-          result += node.__spliceValue;
+          offset += buf.write(node.__spliceValue, offset);
         }
         if (!pretty) {
-          var toReplace = source.slice(startOf(node), endOf(node));
           if (!node.loc || node.loc.start.line === node.loc.end.line) {
-            result += space(toReplace.length);
+            offset += buf.write(space(endOf(node) - startOf(node)), offset);
           } else {
-            var toReplaceLines = toReplace.split(LINE_RX);
-            result += space(toReplaceLines[0].length);
+            var toReplaceLines = sourceBuffer.toString('utf8', startOf(node), endOf(node)).split(LINE_RX);
+            offset += buf.write(space(toReplaceLines[0].length), offset);
             for (var j = 1; j < toReplaceLines.length; j += 2) {
-              result += toReplaceLines[j] + space(toReplaceLines[j + 1].length);
+              offset += buf.write(toReplaceLines[j] + space(toReplaceLines[j + 1].length), offset);
             }
           }
         }
       }
 
-      return (result += source.slice(lastPos));
+      offset += sourceBuffer.copy(buf, offset, lastPos, sourceBuffer.length);
+
+      return buf.toString('utf8', 0, offset);
     },
     generateMap: function() {
```

Comment:
How does this issue translate to source maps? I'm suddenly worried that the source map spec is silent on what "columns" mean, but I'd imagine it's meant to be interpreted more like "code points" than "bytes". So we might need to do some Unicode bookkeeping in `generateSourceMappings` as well.